### ISSUE TITLE
[Bug]:  Plugin generates performance data with value Inf or -Inf

### ISSUE THREAD

**Issue by RincewindsHat** [NETWAYS/go-check#115]
### Please try to fill out as much of the information below as you can. Thank you!

- [X] Yes, I've searched similar issues on GitHub and didn't find any.

### Which version contains the bug?

1.0.2

### Describe the bug

On some occasions (don't know which yet), the plugin produces performance data with the value `Inf` or `-Inf`.
The mode was `query` with, appropriately, a custom query.
Without investigation I am assuming that the retrieved value is empty or invalid in a different way.

As for a solution, I would suggest to filter out such values as I cannot imagine that to be a valid measurement of anything.

**Comment by RincewindsHat** [comment]
no, `0` is a valid value and this would therefore be a false measurement.
Dropping the whole datapoint would be the kind of thing I am looking for.

**Comment by martialblog** [comment]
We could change der String() method on the struct so that it is never printed if it's Inf.

**Comment by martialblog** [comment]
Had a look at the Perfdata type. Think there's a larger discussion to be had. 

This is the code handling the formatting of the Perfdata's value:

```
func formatNumeric(value interface{}) string {
        switch v := value.(type) {
        case float64:
                return check.FormatFloat(v)
        case float32:
                return check.FormatFloat(float64(v))
        case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64:
                return fmt.Sprintf("%d", v)
        case fmt.Stringer, string:
                return fmt.Sprint(v)
        default:
                panic(fmt.Sprintf("unsupported type for perfdata: %T", value))
        }
}
```

Meaning, the value is an interface and we can use strings, or anything that implements the Stringer interface.

Side note: I don't like the `panic` here either. While it is useful for debugging, might be more "robust" (whateverthatmeans) to just not output a Perfdata that can't be formatted. However, this would need to be documented properly otherwise it might be too obscure.

**Comment by martialblog** [comment]
Alas! If only there was a specification as to what a performance data value datatype would be.

**Comment by RincewindsHat** [comment]
As for the `panic`, I think I could not think of a proper way to say "If it's anything else than the above, that should not work". I would prefer to handle this at compile time and abort compilation if possible.
Also, IMHO `string` could be removed, Performance data should always be something in decimal numbers.

**Comment by RincewindsHat** [comment]
Fixed with #116
